modules/pricer/src/test/java/com/opengamma/strata/pricer/fx/BlackFxSingleBarrierOptionProductPricerTest.java

```diff
@@ -0,0 +1,678 @@
+    PointSensitivityBuilder computedCall =
+        PRICER.presentValueSensitivityVolatility(CALL_UKI, RATE_PROVIDER_AFTER, VOL_PROVIDER_AFTER);
+    double computedVegaPut = PRICER.vega(PUT_UKO_BASE, RATE_PROVIDER_AFTER, VOL_PROVIDER_AFTER);
+    PointSensitivityBuilder computedPut =
+        PRICER.presentValueSensitivityVolatility(PUT_UKO_BASE, RATE_PROVIDER_AFTER, VOL_PROVIDER_AFTER);
+    assertEquals(computedVegaCall, 0d);
+    assertEquals(computedCall, PointSensitivityBuilder.none());
+    assertEquals(computedVegaPut, 0d);
+    assertEquals(computedPut, PointSensitivityBuilder.none());
+  }
+
+  //-------------------------------------------------------------------------
+  public void test_currencyExposure() {
+    for (ResolvedFxSingleBarrierOption option : OPTION_ALL) {
+      CurrencyAmount pv = PRICER.presentValue(option, RATE_PROVIDER, VOL_PROVIDER);
+      MultiCurrencyAmount computed = PRICER.currencyExposure(option, RATE_PROVIDER, VOL_PROVIDER);
+      FxMatrix fxMatrix = FxMatrix.builder().addRate(EUR, USD, SPOT + FD_EPS).build();
+      ImmutableRatesProvider provBumped = RATE_PROVIDER.toBuilder().fxRateProvider(fxMatrix).build();
+      CurrencyAmount pvBumped = PRICER.presentValue(option, provBumped, VOL_PROVIDER);
+      double ceCounterFD = pvBumped.getAmount() - pv.getAmount();
+      double ceBaseFD = pvBumped.getAmount() * (SPOT + FD_EPS) - pv.getAmount() * SPOT;
+      assertEquals(computed.getAmount(EUR).getAmount() * FD_EPS, ceCounterFD, NOTIONAL * FD_EPS);
+      assertEquals(-(computed.getAmount(USD).getAmount() - pv.getAmount()) * FD_EPS, ceBaseFD, NOTIONAL * FD_EPS);
+    }
+  }
+
+  public void test_currencyExposure_atExpiry() {
+    for (ResolvedFxSingleBarrierOption option : OPTION_ALL) {
+      CurrencyAmount pv = PRICER.presentValue(option, RATE_PROVIDER_EXPIRY, VOL_PROVIDER_EXPIRY);
+      MultiCurrencyAmount computed = PRICER.currencyExposure(option, RATE_PROVIDER_EXPIRY, VOL_PROVIDER_EXPIRY);
+      FxMatrix fxMatrix = FxMatrix.builder().addRate(EUR, USD, SPOT + FD_EPS).build();
+      ImmutableRatesProvider provBumped = RATE_PROVIDER_EXPIRY.toBuilder().fxRateProvider(fxMatrix).build();
+      CurrencyAmount pvBumped = PRICER.presentValue(option, provBumped, VOL_PROVIDER_EXPIRY);
+      double ceCounterFD = pvBumped.getAmount() - pv.getAmount();
+      double ceBaseFD = pvBumped.getAmount() * (SPOT + FD_EPS) - pv.getAmount() * SPOT;
+      assertEquals(computed.getAmount(EUR).getAmount() * FD_EPS, ceCounterFD, NOTIONAL * FD_EPS);
+      assertEquals(-(computed.getAmount(USD).getAmount() - pv.getAmount()) * FD_EPS, ceBaseFD, NOTIONAL * FD_EPS);
+    }
+  }
+
+  public void test_currencyExposure_afterExpiry() {
+    for (ResolvedFxSingleBarrierOption option : OPTION_ALL) {
+      MultiCurrencyAmount computed = PRICER.currencyExposure(option, RATE_PROVIDER_AFTER, VOL_PROVIDER_AFTER);
+      assertEquals(computed, MultiCurrencyAmount.empty());
+    }
+  }
+
```

Comment:
Add a test on currency exposure comparing vanilla CE with barrier with a very large or very low barrier. This is done for the price indirectly in the Barrier formula repository test, but not for CE.